### ISSUE TITLE
Initial Machine Deployment of User Cluster broken at Multi Seed Setup

### ISSUE THREAD

**Issue by toschneck** [kubermatic/kubermatic#6752]
**User Story**
At the current release `2.16` the multi seed setup user cluster deployment is broken. If you create a user cluster in a "not master" remote seed, the creation of the intial MachineDeployment is failing with the error:
```
failed to create initial MachineDeployment: failed to get SSH keys: failed to get owning project "qffkqxjkkg": Project.kubermatic.k8s.io "qffkqxjkkg" not found
````
![image.png](https://images.zenhubusercontent.com/5b6454d031c5fe7463837d8b/342b6bad-7a83-4805-ae52-28be24f8e46a)

This error occurs no matter if you use a the "User SSH Key Agent" option or not. As the error describes the seed cluster tries to reach the project resources. The project resource is NOT present at the seed cluster, what cause this error.  The corresponding code is:

```
func (r *Reconciler) getSSHKeys(ctx context.Context, cluster *kubermaticv1.Cluster) ([]*kubermaticv1.UserSSHKey, error) {
	var keys []*kubermaticv1.UserSSHKey

	projectID := cluster.Labels[kubermaticv1.ProjectIDLabelKey]
	if projectID == "" {
		return nil, fmt.Errorf("cluster does not have a %q label", kubermaticv1.ProjectIDLabelKey)
	}

	project := &kubermaticv1.Project{}
	if err := r.Get(ctx, ctrlruntimeclient.ObjectKey{Name: projectID}, project); err != nil {
		return nil, fmt.Errorf("failed to get owning project %q: %v", projectID, err)
	}
```

https://github.com/kubermatic/kubermatic/blob/0ae1c4cb0ac8088dab688522f6eaaa79d3d518eb/pkg/controller/seed-controller-manager/initialmachinedeployment/controller.go#L245

The "backend code" for deploying the machinedeployment has been added with the PR #6064.

**Manuel workaround**
A end user could apply a machine deployment manifest directly to the Kubernetes API of the end user.
1. Download kubeconfig
2. Create and apply a valid MachineDeployment yaml 

**Acceptance criteria**
- [ ]  user cluster deployment at seeds can be done without error

**Comment by toschneck** [comment]
I tested the same scenario with the new `weekly-2021-11` as well and it seams, the PR #6708 fixes this issue. I would recommend to backport the Project-Sync-Controller potentially.
Anyway the bug need to get solved in the release `2.16` branch
cc @shaase-ctrl

**Comment by xrstf** [comment]
There is no 2.6 release.

I tried to replicate this on our dev environment and I could not see the problem. Running on e34f174049c563225eb6fbafc66c5c74ca6426b7, before the supposed fix in #6708 was merged. I tried both clusters with and without SSH keys attached to them.

**Comment by xrstf** [comment]
I could reproduce the missing project and then the error, but only when i create a new project. I guess because the new controller temporarily created the new projects and fixed it.

So the bug is confirmed and was most likely introduced when we improved the initial machinedeployment creation.
